parquet2hive/parquet2hive

```diff
@@ -147,6 +147,10 @@ def transform_type(avro):
         record = "struct<{}>".format(fields_decl)
         udf[avro["name"]] = record
         return record
+    elif isinstance(avro, dict) and avro["type"] == "struct":
```

Comment:
It's not exactly the same - the "struct" type doesn't have a "name" field (or at least didn't in this case), so the `udf[...]` piece fails.

Here's the json structure for the row:
```json
{
  "type": "struct",
  "fields": [
    {
      "name": "document_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "client_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "channel",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "normalized_channel",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "country",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "city",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "os",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "os_version",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "os_service_pack_major",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "os_service_pack_minor",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "profile_creation_date",
      "type": "long",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "subsession_start_date",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "subsession_length",
      "type": "long",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "distribution_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "sync_configured",
      "type": "boolean",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "sync_count_desktop",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "sync_count_mobile",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "app_build_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "app_display_version",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "app_name",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "app_version",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "timestamp",
      "type": "long",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "env_build_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "env_build_version",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "env_build_arch",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "e10s_enabled",
      "type": "boolean",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "e10s_cohort",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "locale",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "active_experiment_id",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "active_experiment_branch",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "reason",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "timezone_offset",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "plugin_hangs",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "aborts_plugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "aborts_content",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "aborts_gmplugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crashes_detected_plugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crashes_detected_content",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crashes_detected_gmplugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_attempt_main",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_attempt_content",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_attempt_plugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_success_main",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_success_content",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "crash_submit_success_plugin",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "active_addons_count",
      "type": "long",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "flash_version",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "vendor",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "is_default_browser",
      "type": "boolean",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "default_search_engine_data_name",
      "type": "string",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "loop_activity_open_panel",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "loop_activity_open_conversation",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "loop_activity_room_open",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "loop_activity_room_share",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "loop_activity_room_delete",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "devtools_toolbox_opened_count",
      "type": "integer",
      "nullable": true,
      "metadata": {}
    },
    {
      "name": "search_counts",
      "type": {
        "type": "array",
        "elementType": {
          "type": "struct",
          "fields": [
            {
              "name": "engine",
              "type": "string",
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "source",
              "type": "string",
              "nullable": true,
              "metadata": {}
            },
            {
              "name": "count",
              "type": "long",
              "nullable": true,
              "metadata": {}
            }
          ]
        },
        "containsNull": true
      },
      "nullable": true,
      "metadata": {}
    }
  ]
}

```